dapps/marketplace/src/utils/growthTools.js

```diff
@@ -209,3 +212,44 @@ export function getContentToShare(action, locale) {
 
   return translation ? translation.text : action.content.post.tweet.default
 }
+
+export function setReferralCode(code) {
+  if (!code) {
+    delete localStorage.partner_referral_code
+    return
+  }
+
+  if (localStorage.partner_referral_code === code) {
+    return
+  }
+
+  localStorage.partner_referral_code = code
+}
+
+export function hasReferralCode() {
+  return !!localStorage.partner_referral_code
+}
+
+export function getReferralReward(campaign) {
+  const referralCode = localStorage.partner_referral_code
+
+  const actions = get(campaign, 'actions', [])
+
+  if (!referralCode || !actions.length) {
+    return null
+  }
+
+  const action = actions.find(
+    action =>
+      action.type ===
+      (referralCode.startsWith('op') ? 'PartnerReferral' : 'Referral')
+  )
+
+  const reward = get(action, 'reward.amount')
```

Comment:
This is the potential reward, not the reward earned.  This Reward will also show up for users who did not arrive via the referral.  The amount here will likely always be 0 as well since the potential reward is variable.

However, I don't think `rewardEarned` will be set until device registration and the reward is unlocked from the 3 prerequisites(profile, e-mail, etc).

Reply:
Might need to load `campaigns.json` for this information...

Reply:
Thanks. Will update this.

Reply:
Working on a fix for this, but not holding up this PR.